.github/workflows/codemod_publish.yml

```diff
@@ -1,107 +1,82 @@
-name: Codemod publish
+# yaml-language-server: $schema=https://json.schemastore.org/github-workflow.json
+
+# For more information see: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow
+
+name: Publish Codemod
 
 on:
-  workflow_dispatch:
   push:
-    paths:
-      - "recipes/**"
-    branches:
-      - main
-
-permissions:
-  contents: read
+    tags:
+      - 'v*@*'  # Match tags like v1.0.0@my-codemod
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Tag to publish (e.g., v1.0.0@my-codemod)'
+        required: true
+        type: string
 
 jobs:
-  paths-filter:
-    permissions:
-      contents: read
-      pull-requests: read
-    name: Check for .codemodrc.json files changes
+  publish:
     runs-on: ubuntu-latest
-    outputs:
-      codemods: ${{ steps.filter.outputs.codemods }}
-      codemods_files: ${{ steps.filter.outputs.codemods_files }}
+
     steps:
-      - name: Harden the runner (Audit all outbound calls)
-        uses: step-security/harden-runner@6c439dc8bdf85cadbbce9ed30d1c7b959517bc49
-        with:
-          egress-policy: audit
+      - name: Checkout code
+        uses: actions/checkout@v4
 
-      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
-      - uses: dorny/paths-filter@de90cc6fb38fc0963ad72b210f1f284cd68cea36
-        id: filter
-        name: Filter codemods
-        with:
-          list-files: json
-          filters: |
-            codemods:
-              - '**/.codemodrc.json'
+      - name: Extract version and codemod name
+        id: extract
+        run: |
+          if [ "${{ github.event_name }}" == "workflow_dispatch" ]; then
+            TAG="${{ github.event.inputs.tag }}"
+          else
+            TAG="${GITHUB_REF#refs/tags/}"
```

Comment:
It's not possible for a non-member to create a tag, right? (I'm thinking this could be a potential attack vector)

Reply:
Correct. Only users with `write` access to this repository can add tags.

Reply:
only people with write role can create a tag.